src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java

```diff
@@ -146,4 +150,20 @@
      * @return {@code true} if this AST has any children.
      */
     boolean hasChildren();
+
+    /**
+     * Get list of tokens on COMMENTS channel to the left of the
+     * current token up to the preceding token on the DEFAULT_TOKEN_CHANNEL.
+     *
+     * @return list of comment tokens
+     */
+    List<Token> getHiddenBefore();
```

Comment:
why we need to extend API ?
"hidden" is nuances of implementation of parser. 
AST is just a tree, it only have nodes.

Reply:
Done.